dogstatsd.mli

```diff
@@ -0,0 +1,140 @@
+module Base : sig
+  module Tag : sig
+    type t = string * string (** (tag-name, value) *)
+  end
+
+  (** https://docs.datadoghq.com/developers/dogstatsd/#metrics *)
+  module Metric : sig
+    module Counter : sig
+      type t =
+        [ `Increment
+        | `Decrement
+        | `Value of int ]
+    end
+
+    type typ =
+      [ `Counter of Counter.t
+      | `Gauge of float
+      | `Timer of float
+      | `Histogram of int
+      | `Set of string ]
+
+    type t =
+      { metric_name : string
+      ; metric : typ
+      ; sample_rate : float option
+      ; tags : Tag.t list }
+  end
+
+  (** https://docs.datadoghq.com/developers/dogstatsd/#service-checks *)
+  module ServiceCheck : sig
+    type status =
+      [ `Ok
+      | `Warning
+      | `Critical
+      | `Unknown ]
+
+    type t =
+      { name : string
+      ; status : status
+      ; timestamp : int option
+      ; hostname : string option
+      ; tags : Tag.t list
+      ; message: string option }
+  end
+
+  (** https://docs.datadoghq.com/developers/dogstatsd/#events *)
+  module Event : sig
+    type priority = [ `Normal | `Low ]
+
+    type alert =
+      [ `Error
+      | `Warning
+      | `Info
+      | `Success ]
+
+    type t =
+      { title : string
+      ; text : string
+      ; timestamp : int option
+      ; hostname : string option
+      ; aggregation_key : string option
+      ; priority : priority option (* defaults to Normal *)
+      ; source_type_name : string option
+      ; alert_type : alert option (* defaults to Info *)
+      ; tags: Tag.t list }
+  end
+end
+
+(* TODO: replace IO with {Async, Lwt, Sync} *)
+module IO : sig
+
+  module Metric : sig
+    (** A couple of examples of how to send metrics:
+
+        * Async.Metric.send
+            ~tags:["tag1", "val1"; "tag2","val2"]
+            ~sample_rate:0.1
+            (`Counter `Increment) ["application.page_views"]
+        * Async.Metric.send
+            (`Set "some-user-id") ["application.user_ids"]
+
+    *)
+
+    val t_send : Base.Metric.t -> unit
+    val send
+      : ?tags:Base.Tag.t list
+      -> ?sample_rate:float
+      -> Base.Metric.typ
+      -> string list
```

Comment:
Why can this take multiple names but `ServiceChek.t_send` can't?

Reply:
I added this for convenience since it seems like you might send multiple metrics at the same time but I can remove it and we can revisit if it proves to be useful for the user. And I didn't include it for ServiceCheck since I couldn't think of a time where multiple service checks would need to be sent with the same message.

Reply:
ah, makes sense